RED-119 add github ci

TieDyedSheep opened the pull request:
DO NOT MERGE without approval from Andrew!

thantsintoe committed:
selectNeighbors function improvement

Glitch18 committed:
feat: TS port of getCorrespondingNodes and verfifyCorrespondingSender

thantsintoe committed:
initial implementation of factTellCorrespondingNodes and factValidateCorrespondingTellSender functions

thantsintoe committed:
fix bug in filtering the corresponding nodes and validating them. added temp logs (which will be removed later)

thantsintoe committed:
refactor factValidateCorrespondingTellSender()

thantsintoe committed:
proposed a new way for verifyCorrespondingSender function

thantsintoe committed:
set correct value for correspondingGlobalOffset

thantsintoe committed:
verifyCorrespondingSender function update for handle edge case

thantsintoe committed:
implement `factTellCorrespondingNodesFinalData` and `factValidateCorrespondingTellFinalDataSender` using FACT algorithm

Glitch18 committed:
feat: Implement factSendCorrespondingCachedAppData

Glitch18 committed:
feat: implement factValidateCorrespondingCachedAppDataSender

Glitch18 committed:
fix: remove unused variable

Glitch18 committed:
fix: remove not required async tag

achal-singh committed:
BLUE-119: Added useNewPOQ and devPublicKeys to the /netconfig endpoint

Glitch18 committed:
fix: factValidateCorrespondingCachedAppDataSender

Glitch18 committed:
feat: only use factValidation when flag is active

thantsintoe committed:
move neighbor execution node check to outside of factValidateCorrespondingTellSender function

thantsintoe committed:
refactor factTellCorrespondingNodesFinalData for better perf

thantsintoe committed:
code refactor after rebases

Glitch18 committed:
feat: add fact function call

Glitch18 committed:
fix: index calculation in fact send cached app data

Glitch18 committed:
fix: target groups for send cached app data

thantsintoe committed:
fix a bug in calculating the cachedAppData corresponding nodes

jairajdev committed:
Convert the corresponding-test.js test script to unit test

rename file

jairajdev committed:
type fix in pruneNetworkChangeQueue dapp function

jairajdev committed:
updated to remove old entries from appliedConfigChanges that are no longer in the changes list

Glitch18 committed:
feat: implement configs for poqo

Glitch18 committed:
feat: implement poqo tryProduceReceipt

Glitch18 committed:
feat: implement consensing step for poqo

Glitch18 committed:
feat: voteHash calculation for POQo

Glitch18 committed:
feat: implement POQo send vote loop

Glitch18 committed:
feat: gossip handlers related to poqo

Glitch18 committed:
fix: gossip forwarding for poqo receipt

Glitch18 committed:
feat: implement common origin non burst gossip

Glitch18 committed:
fix: calculations in sendCorrespondingCachedAppData

Glitch18 committed:
fix: poqo handler for sendCorrNodesFinalData

Glitch18 committed:
fix: tryProduceReceipt for POQo

thantsintoe committed:
tx timestamp caching improvements for robustness

Glitch18 committed:
feat: add log for tracking sharing of cachedAppData

Glitch18 committed:
feat: add logs to POQo endpoints

Glitch18 committed:
feat: populate receiveAppliedReceipt2 in POQo

thantsintoe committed:
lastVoteReceivedTimestamp fix for old POQ

afostr committed:
also update time for other added votes

Glitch18 committed:
fix: edge case causing missing receipt

Glitch18 committed:
refac: prepone check for null cache data

Glitch18 committed:
feat: POQo switch to requiredVotesPercentage config

Glitch18 committed:
feat: add new conditions for initiating gossip

thantsintoe committed:
collected data fix. use newer accounts if we receive the same account multiple times. prioritise to use our own data if we have it

thantsintoe committed:
if ourVote mismatches the winning receipt, go to "await final data" and use the data attached. No need to repair tx.

ahmxdiqbal committed:
add flag to track if we have patched since going active

add cycle we finished syncing to report

added guard for setting stillNeedsInitialPatchPostActive flag

add bool to radix report to indicate if the radix has recently synced

set recentRuntimeSync in simpler way + account for wrapped ranges

mssabr01 committed:
updated config settings to be closer to prod ready

thantsintoe committed:
remove duplicate collectedData check in "queueEntryAddData"

thantsintoe committed:
add a flag to control to reject shared data if we cover it. set to "false" by default

thantsintoe committed:
Add rejectSharedDataIfCovered flag to binary endpoint as well

Glitch18 committed:
fix: calculate numVotes in POQo

afostr committed:
adjust fix flags for POQo

abdulazeem-tk4vr committed:
GOLD-172 Migrating PoQo endpoints to Binary (#172)

* feat: binary migration of popqo endpints

* PoQo Send Receipt Binary

Binary Migration of PoQo Send Receipt : Handler, Serializer and Deserializer Additions

* feat: Add poqoSendReceiptBinary configuration option

The `poqoSendReceiptBinary` configuration option is added to enable binary migration of the PoQo Send Receipt feature. This commit includes the necessary additions to the handler, serializer, and deserializer for the binary migration.

* refactor: Update internal route for sending receipt in TransactionConsensus.ts

* formatting endpoint position

* feat: Add sending final receipt gossip to binary handler in TransactionConsensus.ts

* refactor: Improve error handling in poqoSendVoteBinary handler

* feat: Fix error handling and improve binary migration in poqo  data and receipt

* refactor: Update poqoSendReceiptReq deserialization logic

Refactor the deserialization logic in the `deserializePoqoSendReceiptReq` function in the `PoqoSendReceiptReq.ts` file. Simplify the code by removing the unnecessary `else` block and handling the `confirmOrChallenge` property more efficiently.

* fix: redundant binary endpoint call

* fix: redundant serde functions and enable bin serializtion

---------

Co-authored-by: Kaung Myat Thu <2962000kaungmyatthu@gmail.com>
Co-authored-by: Atharva <atharvakn@gmail.com>

S0naliThakur committed:
error handling for all external endpoints

S0naliThakur committed:
error handling for all external endpoints

S0naliThakur committed:
update endpoint-exception counter message

S0naliThakur committed:
update endpoint-exception counter message

Commit:
GOLD-164: Signature majority validation for non locally produced receipts (#162)

* Added signature majority validation

* refac: minor changes in verifyAppliedReceipt

* feat: add sig verification for binary endpoints

* feat: signature verification for appending vote hash

* fix: poqo endpoint sig verification

---------

Co-authored-by: Atharva <atharvakn@gmail.com>

Glitch18 committed:
feat: debug-fails DO NOT MERGE

jairajdev committed:
Move the receipt sig verification check before processing the accounts data in the poqo-data-and-receipt handler (#177)

* Move the receipt sig verification check before processing the accounts data in the poqo-data-and-receipt

* feat: prepone fact CT sender check

---------

Co-authored-by: Atharva <atharvakn@gmail.com>

TieDyedSheep committed:
RED-119 add github ci

fixing issue comment Pr review stuff

paulgs9988 reviewed (dismissed):
Similar issue to the review I left on the Shardeum ci.yml PR:

We have "build:dev" but not "build" on the shardus-core repo so we get a failure from the github automation "QA merge checks", but unrelated to this guy here.

afostr committed:
fix for corresponding tell.  improved logs and counters

afostr committed:
support to allow awaiting data to bail when there is a valid receipt.   improved counters.  disable some unused code

config to control requesting data while waiting

afostr committed:
Bump version to 2.12.30-59

Glitch18 committed:
fix: disable filtering of receiving nodes for tellCorrespondingNodes

afostr committed:
additional logs and counters cleanup/fixes

afostr committed:
missed this

afostr committed:
crash fixes

urnotsam committed:
Add new debug endpoint to query for transactions in queue based on time and state

Glitch18 committed:
fix: comparison bug in factTellCorrespondingNodes

squash and remove

jairajdev committed:
Updated `forwardToLuckyNodesNonceQueue` flag to true

afostr committed:
wip fixes for unwrapping sender indexes

thantsintoe committed:
calculate and attach a flag to queueEntry if the sender group is wrapped and our node is on the left side of the wrapped range

urnotsam committed:
add cycle of recent run time sync

afostr committed:
add disabled diagnostic tx cancel.  add diagnostic forward to nonce. forward to local multi option

fix

afostr committed:
Bump version to 2.12.30-60

Glitch18 committed:
fix: bug in binary serialization for appliedReceipt2

tanuj-shardeum committed:
feat: poqo-request-data implementation (#175)

Glitch18 committed:
feat: implement using unwrapped index in tellFinallData

abdulazeem-tk4vr committed:
refactor: update TransactionConsensus to use readableReq instead of payload

Glitch18 committed:
feat: reenable filtering in factTellCorrespondingNodes

Glitch18 committed:
fix: binary serialization for AppliedReceipt2

jairajdev committed:
Added to send txGroupCycle along with the receipt to debug txGroupCycle mismatch issue

chrypnotoad committed:
add build script for compile

chrypnotoad commented on package.json:
```diff
@@ -34,6 +33,7 @@
     "release:move": "npm-run-all release:move:*",
     "build:dev": "npm-run-all clean compile:ts dev:move",
     "build:release": "npm-run-all clean prep compile release:move",
+    "build": "npm run compile",
```
don't need this anymore

chrypnotoad commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,77 @@
+    with:
+      node-version: ${{ vars.NODE_VERSION }}
+      lint-required: ${{ vars.IS_LINT_REQUIRED == 'true' }}
+      format-check-required: ${{ vars.IS_FORMAT_CHECK_REQUIRED == 'true' }}
+      apply-patches-required: ${{ vars.IS_APPLY_PATCHES_REQUIRED == 'true' }}
```
should be false

TieDyedSheep committed:
added CODEOWNERS file

correct quotes on lint script

update job names so their purpose is more clear

remove unnecessary build script

chrypnotoad commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,77 @@
+      contents: write
+    with:
+      node-version: ${{ vars.NODE_VERSION }}
+      lint-required: ${{ vars.IS_LINT_REQUIRED == 'true' }}
+      format-check-required: ${{ vars.IS_FORMAT_CHECK_REQUIRED == 'true' }}
```
```suggestion
      lint-required: ${{ vars.IS_LINT_REQUIRED == 'false' }}
      format-check-required: ${{ vars.IS_FORMAT_CHECK_REQUIRED == 'false' }}
```

chrypnotoad commented on CODEOWNERS:
```diff
@@ -0,0 +1,13 @@
+/.github/ @shardeum/systems-and-automation
+
+# Specific file rules
+# README.md @shardeum/team-name
```
```suggestion
# README.md @shardeum/team-name

```

chrypnotoad commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,77 @@
+    with:
+      node-version: ${{ vars.NODE_VERSION }}
+      lint-required: ${{ vars.IS_LINT_REQUIRED == 'true' }}
+      format-check-required: ${{ vars.IS_FORMAT_CHECK_REQUIRED == 'true' }}
+      apply-patches-required: ${{ vars.IS_APPLY_PATCHES_REQUIRED == 'true' }}
```
```suggestion
      format-check-required: ${{ vars.IS_FORMAT_CHECK_REQUIRED == 'false' }}
      apply-patches-required: ${{ vars.IS_APPLY_PATCHES_REQUIRED == 'false' }}
```

chrypnotoad commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,77 @@
+      contents: write
+    with:
+      node-version: ${{ vars.NODE_VERSION }}
+      lint-required: ${{ vars.IS_LINT_REQUIRED == 'false' }}
+      format-check-required: ${{ vars.IS_FORMAT_CHECK_REQUIRED == 'false' }}
```
```suggestion
      lint-required: ${{ vars.IS_LINT_REQUIRED == 'true' }}
      format-check-required: ${{ vars.IS_FORMAT_CHECK_REQUIRED == 'true' }}
```

chrypnotoad commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,77 @@
+      contents: write
+    with:
+      node-version: ${{ vars.NODE_VERSION }}
+      lint-required: ${{ vars.IS_LINT_REQUIRED == 'true' }}
+      format-check-required: ${{ vars.IS_FORMAT_CHECK_REQUIRED == 'false' }}
+      apply-patches-required: ${{ vars.IS_APPLY_PATCHES_REQUIRED == 'false' }}
+      unit-tests-required: ${{ vars.IS_UNIT_TESTS_REQUIRED == 'true' }}
```
```suggestion
      lint-required: ${{ vars.IS_LINT_REQUIRED == 'true' }}
      format-check-required: ${{ vars.IS_FORMAT_CHECK_REQUIRED == 'true' }}
      apply-patches-required: ${{ vars.IS_APPLY_PATCHES_REQUIRED == 'true' }}
      unit-tests-required: ${{ vars.IS_UNIT_TESTS_REQUIRED == 'true' }}
```

chrypnotoad committed:
Apply suggestions from code review
checks should be true to use env variables

chrypnotoad commented:
/describe

TieDyedSheep committed:
remove outdated patch that's breaking the CI

TieDyedSheep commented:
/describe

thantsintoe committed:
implement deterministic txGroupCycle and txGroup calculation. Put behind "deterministicTXCycleEnabled" flag. Set false by default

thantsintoe committed:
prevent "cycleToRecordOn" from using deterministic txGroupCycle

thantsintoe committed:
set reduceTimeFromTxTimestamp:5000 by default

achal-singh committed:
chore: fixes to make /debug-neverGoActive usable

chrypnotoad commented:
DO NOT MERGE. Will be merged by Andrew.

thantsintoe committed:
updated "queueEntryGetConsensusGroup" and "queueEntryGetConsensusGroupForAccount" to  consider deterministicTXCycleEnabled flag

thantsintoe committed:
fix to exclude current cycle from txGroupCycle calculation. Better logs and counters

thantsintoe committed:
added fallbackToCurrentCycleFortxGroup for deterministicTXCycleEnabled feature

afostr committed:
fix and logging for run-away forward to lucky TX

afostr committed:
Bump version to 2.12.30-61

jairajdev committed:
Added txGroupCycle and uniqueKeys in the debug-queue-items

jairajdev committed:
Added debug-queue-item query by txId

jairajdev committed:
Added full txId in the debug queue item

PudgyPug committed:
update settings for new rc

afostr committed:
Bump version to 2.12.30-62

TieDyedSheep committed:
added format-check and fix

cleanup gitlab.ci

thantsintoe committed:
use "concatCorrespondingTellUseUnwrapped" flag whether to concat extra corresponding indices or not

thantsintoe committed:
fact tell functions should use shard data from deterministic txGroupCycle

thantsintoe committed:
extra useful debug info, counters and logs

thantsintoe committed:
set "filterReceivingNodesForTXData" to false by default to false for safety over optimization

urnotsam committed:
get current max queueentry time in report

achal-singh committed:
fix: emitSyncTimeoutEvent call position fixed

afostr committed:
updated debug info

thantsintoe committed:
always concat extra wrap index and the receiver is okay with both indices.

afostr committed:
Bump version to 2.12.30-63

afostr committed:
fix for failed TXs not clearing queue after consensus

afostr committed:
add in collected votes

afostr committed:
additional corrections to receipt generation

afostr committed:
Bump version to 2.12.30-64

afostr committed:
Bump version to 2.12.30-65

tanuj-shardeum committed:
Add simpleDebugStr to debug-queue-items endpoint report

jairajdev committed:
maxCyclesShardDataToKeep config

Added missing debug configs in the netconfig endpoint

change the description

afostr committed:
add .?
